### DISCUSSION TITLE
Why there are creases with “ surface from curves“

### DISCUSSION THREAD

**Discussion by 369386973** [nortikin/sverchok#4794] [Q&A]
![20221205154817](https://user-images.githubusercontent.com/101614185/205582651-e9075cba-6a51-4ee0-8f71-4cf8a00150be.jpg)
As the picture shows，there is a crease on the surface when I use the node “surface from curves” with three circles.And when I use the node “dual mesh”, it is obviously disconnected.
Why is it like this？Thanks for reply！

**Comment by portnov** [comment]
Try to use "merge by distance" before "dual mesh".

**Reply by 369386973** [reply]
> Try to use "merge by distance" before "dual mesh".
![20221205183405](https://user-images.githubusercontent.com/101614185/205616636-f63471ee-50c2-48f5-ae20-8e0b267dabd7.jpg)


Thank you！It‘s helpful！
But when I use the node“Bend along surface field”, it disconnects again.

**Reply by portnov** [reply]
Yes. Unfortunately, at least at the moment, Sverchok does not have concept of "closed surface". It can generate a surface, one edge of which coincides with another, but that's all: it will not understand such coincidence as continuation of surface.

**Comment by 369386973** [comment]
So can I use any other nodes to complete this step？As long as the edges can get together normally.

**Comment by rendetto** [comment]
I think that edges are joint normally - this is the way surfaces work. The only problem is with the vertices of the surface-mesh-visualizer that are not merged. You can do it manually.
Can you use a mesh for this base object instead? Meshes have the properties you seem to need and are much easy to handle.

But if you need surfaces then you need to cheat - use faces (trim surfaces) instead. You need to construct the surfaces and cut it in a way that natural edges are not included/visible in the final object. Then you won't have seams.

**Reply by 369386973** [reply]
At first  I wanted to use a mesh , but I didn't find a mesh node like “Bend Along Surface Field”.

And which node can help me cut the surfaces in the way you said？

**Reply by rendetto** [reply]
> And which node can help me cut the surfaces in the way you said？

From Solids - "Face from Surface" is the node.

**Reply by 369386973** [reply]
> Can you use a mesh for this base object instead?

thx！
One more question ：Is there any mesh node like  “Bend Along Surface Field”.

**Reply by rendetto** [reply]
I don't really know. It depends more of what is your goal, than the tool itself. 
There are many ways to achieve something with Sverchok. For example you can do something like that:

![voro_on_mesh](https://user-images.githubusercontent.com/66558924/206300683-205ebd70-b8ff-444b-a47f-53908d702cec.jpg)

You'll always need to tinker it, and that's the fun part :)
Cheers

**Reply by 369386973** [reply]
thx again！That's what I want
